### PR TITLE
Feature/stable diffussion blog

### PR DESCRIPTION
**Pull request by rjczanik**
A blog entry for gen-ai image renderer stable-diffusion. 
Not sure how to force images to stick to defined pixel widths as defined in .md file.

### CODE REVIEW

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: "Using generative AI models to make a scetch from a photo"
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: I'm guessing you mean "sketch", right?

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+## What is generative AI and why use it?
+
+In this blog post generative AI is introduced, specifically one of the most popular prompt-to-image packages: Stable diffusion.
+First of all, the question is what is generative AI: Generative AI is software that uses some form of input from a user to generate content using a deep learing model.
+The input is usually given as text, in the form of a prompt which specifies the description of the type of content that you want to generate.
+Output generated is usually in some form of media that can be viewed or intepreted by a person in either the form of text, images or in some of the latest generation of Gen-AI models video.
+
+Now with a bit of a definition on what Gen-AI is what it does, I am going to give a bit of a summary on how fun it is to play around with some prompts on one of the most popular image-generating AI: Stable diffusion.
+
+## What is stable diffusion, who developed it, and why should you care
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: Oxford comma

> What is stable diffusion, who developed it and why should you care?

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+One of the most popular image Gen-AI packages is called stable diffusion, which takes in text prompts and generates images according to the prompt from a chosen specific model.
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: When we're naming a piece of technology like Stable Diffusion, I think it's best to capitalize it consistently throughout the entire article.

Comment (LFSb) [reply content/blog/gen-ai-local-stable-diffusion.md]: The same counts for referring to Open Source, and stuff like that :)

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+This type of technology is under my opinion as revolutionary as the transition from film to digital photography, or even the invention of the printing press.
+
+## Where can you get it, and how to use it
+
+One can very easily get the model and code for stable diffusion from Stability-AI's [website](https://stability.ai/).
+The code is not too difficult to understand, but if one does not want the retrain a model or generate custom scripts here is a web-app that is user fiendly on github, [AUTOMATIC1111](https://github.com/AUTOMATIC1111/stable-diffusion-webui).
+This specific web-app does all the config that you need for a GPU, if you are running on Mac or Linux all of the nececary file installations are done when you execute the webui.sh script. Note that this was not tested on Windows, but it looks as if it will also run there. Moreover, if one has an AMD based GPU then on Linux you would first have to install [ROCM](https://www.amd.com/en/products/software/rocm.html) and on Windows you would have to use Direct-ML, which can both be a challenge to setup.
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: typo: `nececary` = necessary

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+An alternative popular option to AUTOMATIC1111's interface would be the [Comfy-UI](https://github.com/comfyanonymous/ComfyUI) web-app, which was not tested for this blog.
+
+### Creating an image from a prompt
+
+The first step to start playing around with stable-diffussion, would be to download the code from github and execute the webui.sh script in a virtual Python environment, at the time of writing the latest version of the interface required Python 3.10.6.
+Upon the execution of the webui.sh script the first time all of the necerary dependancies are installed and the basic model, stable diffusion 1.5 is downloaded from hugginface's website.
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: same typo `nececary`, also dependancies should be dependencies. Also, shouldn't it be "huggingface"?

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+The user-interface is quite intuitive, below the model selection pane there are tabs, of which the first two are only relevant to this blog; *txt2img* where a prompt generates an image, and *img2img* where one can give an image and a prompt as input and a new image is generated.
+And then in the pane below the tabs one sees the tuning parameters for the model that are used for the render and to the right the rendered image is displayed at the end of the render.
+
+{{<img src="/img/blog/model-name.png" class="img-float" title="Selection of the model's checkpoint"  width="350">}}
+
+For this specific excercise I wanted to see what a typical model would render if given the prompt for an Anime style image of a machine learning engineer (or more precisely how it would render an anime image of myself.)
+I started playing around with the baseline model, and quikly discovered that it does not do very well trying to render anime style of images. So after searching around online for the best models for this type of output, I settled on the *protogenV22Anime* model.
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: Typo, quikly should be quickly

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+So I decided to use the following image of myself and turn it into an anime style image:
+
+{{<img src="/img/blog/original.jpeg" class="img-float" title="A selfie, cropped to the *512 x 512* image size that would best work for an image-to-image render using stable diffusion."  width="450">}}
+
+Before altering the image it was cropped to the ideal size of 512 x 512 pixles, this is also the same size of images rendered by prompt only due to memory and performance demands.
+The basic stable diffussion model did not render good results of and anime version of the input image,  the *anything-V3* model was chosen due to it's broad range of capabilities of image rendering.
+
+{{<img src="/img/blog/model-name.jpeg" class="img-float" title="The choice of the model used for altering a photo to an anime version."  width="350">}}
+
+One thing that is interesting in the naming of the model name shown above, is that the model is a pruned version of *anything-V3* is a slimmed down version which has any unececary artifacts pruned out.
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: typo: `unececary` = unnecessary

File: content/blog/gen-ai-local-stable-diffusion.md [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+{{<img src="/img/blog/img2img.jpeg" class="img-float" style="float:left" title="The user interface of AUTOMATIC1111's webapp when altering an existing image according to a prompt."  width="750">}}
+
+For altering an image the render parameters had the following values: *CFG-Scale=8, Denoising strenght=0.5* and a batch of roughly 65 images were rendered.
+These values were chosen after reading up online a bit and doing a bit of *prompt and pray*-ing, which can be a bit of a time consuming exercise.
+The two best resulting images are:
+
+{{<img src="/img/blog/anime1.jpeg" class="img-float" style="float:left; margin-right: 10px;" title="The best version of an anime image of the photo uploaded."  width="400">}}
+{{<img src="/img/blog/anime2.jpeg" class="img-float" style="float:left" title="A tougher version with sharper edges of the anime image."  width="400">}}
+
+These images are relativly close to the original photo and it is quite impressive to someone like myself who has not a strong artistic talent.
```
Comment (LFSb) [inline content/blog/gen-ai-local-stable-diffusion.md]: typo `relativly` = relatively

**Review by LFSb — COMMENTED** [review]
Very shallow read-through, mostly looking for typo's for now.

File: content/blog/gen-ai-local-stable-diffusion.md (hunk 2 above)
Comment (rjczanik) [reply content/blog/gen-ai-local-stable-diffusion.md]: Not removing my Oxford commas :)

**Review by LFSb — APPROVED** [review]
LGTM

**Pull request merged**